### DISCUSSION TITLE
Help

### DISCUSSION THREAD

**Discussion by EdwardDanielSalazar** [konwektor/BlueRetro#23] [Q&A] [answered]
Good day, I am compiling the code, and everything runs normally, it creates the 4 .bins, but when I upload them to the esp32, it does not work, it is as if it does not start, should I configure something additional? Could you help me on what? Maybe, with release 1.9 it works normally, forgive my English

**Comment by konwektor** [comment, marked as answer]
Hi there. 
without more details  hard to say what is wrong, - compilation, esp32, connections between arduino and esp33-  but ...
You are compiling code as I understand. - again  without info i could only guest, what idf, system, compiler.

I am always compiling eveything in ESP IDF under windws 10 in command window (cmd.exe - with administrator privileges)
For compiling simple command : idf.py build
After compiling for upload to esp32 : idf.py  -p com5 flash (in my computer i got esp32 on com5 - check in devices manager)
Afrer upload Iam always starting monitor to see log  from esp32, and I think this is what you need to start ALWAYS after uploading code.
Check on what com you got your esp32 - lets say comX

Run : idf.py -p comX monitor
Check messages, and let me know -I try to help.

**Reply by EdwardDanielSalazar** [reply]
If all this in order compiles and performs the normal flash, I manage it the same from the windows idf, with the sdkconfig.defaults, without modifying it works without problem, but the controller does not work, when adding the ogx360 lines and when doing the flash the esp32 no longer works, it's like it restarts very quickly, I think it's something with the sdk configs, could you compare yours?

**Reply by EdwardDanielSalazar** [reply]
TTYbWWJsTj9HXmvvgFmmXkqqKEaQrEWRJb

--- WARNING: GDB cannot open serial ports accessed as COMx
--- Using \\.\COM5 instead...
--- esp-idf-monitor 1.3.4 on \\.\COM5 921600 ---
--- Quit: Ctrl+] | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H ---
W (28) boot: Unicore bootloader
I (270) cpu_start: Unicore app
I (270) cpu_start: Pro cpu up.
I (271) cpu_start: Single core mode
I (278) cpu_start: Pro cpu start user code
I (279) cpu_start: cpu freq: 240000000 Hz
I (279) cpu_start: Application information:
I (279) cpu_start: Project name:     BlueRetro
I (280) cpu_start: App version:      1
I (280) cpu_start: Compile time:     Mar 22 2024 15:59:52
I (281) cpu_start: ELF file SHA256:  27d79b0957e136c0...
I (282) cpu_start: ESP-IDF:          v5.1.2-dirty
I (282) cpu_start: Min chip rev:     v1.0
I (283) cpu_start: Max chip rev:     v3.99
I (283) cpu_start: Chip rev:         v1.0
I (284) cpu_start: App cpu up.
I (285) heap_init: Initializing. RAM available for dynamic allocation:
I (286) heap_init: At 3FFAFF10 len 000000F0 (0 KiB): DRAM
I (286) heap_init: At 3FFB6388 len 00001C78 (7 KiB): DRAM
I (287) heap_init: At 3FFB9A20 len 00004108 (16 KiB): DRAM
I (288) heap_init: At 3FFCD448 len 00012BB8 (74 KiB): DRAM
I (288) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (289) heap_init: At 3FFE4350 len 00003A48 (14 KiB): D/IRAM
I (290) heap_init: At 3FFE7E28 len 000181D8 (96 KiB): D/IRAM
I (291) heap_init: At 40078000 len 00008000 (32 KiB): IRAM
I (292) heap_init: At 4009F568 len 00000A98 (2 KiB): IRAM
I (292) heap_init: At 3FF80068 len 00001F98 (7 KiB): RTCRAM
I (294) spi_flash: detected chip: generic
I (294) spi_flash: flash io: dio
I (295) esp_core_dump_uart: Init core dump to UART
I (295) esp_core_dump_uart: Init core dump to UART
I (296) app_start: Starting scheduler on CPU0
I (296) main_task: Started on CPU0
I (296) main_task: Calling app_main()
# App main at 400834EC
# App init at 400834FC
# APP_CPU RESET: 0
# APP_CPU CLKGATE: 0
# APP_CPU STALL: 0
# APP_CACHE_CTRL: 00000030
# APP CPU STACK PTR: 3FFBCEB0
# Hardcoded system : 24: OGX360
config_load_from_file: Upgrading cfg v2 to v3
Guru Meditation Error: Core  0 panic'ed (LoadStoreAlignment). Exception was unhandled.

Core  0 register dump:
PC      : 0x400940c8  PS      : 0x00060033  A0    x0927 1    xfb30
0x400940c8: vTaskSwitchContext at C:/Espressif/frameworks/esp-idf-v5.1.2/components/freertos/FreeRTOS-Kernel/tasks.c:3743
 (inlined by) vTaskSwitchContext at C:/Espressif/frameworks/esp-idf-v5.1.2/components/freertos/FreeRTOS-Kernel/tasks.c:3673

A   :00000 A     xfc24 8    x088d 9    x0001
A0  :000004 1    0x3ffc4344  A12     : 0x3ffc4328  A13     : 0x400828ad
0x400828ad: _xt_handle_exc at C:/Espressif/frameworks/esp-idf-v5.1.2/components/freertos/FreeRTOS-Kernel/portable/xtensa/xtensa_vectors.S:811

A14     :����L3ff000dc  A15     : 0x000000R

**Comment by EdwardDanielSalazar** [comment]
W (58) boot: Unicore bootloader
W (58) boot.esp32: PRO CPU has been reset by WDT.
W (58) boot.esp32: WDT reset info: PRO CPU PC=0x400803c0
0x400803c0: _DoubleExceptionVector at C:/Espressif/frameworks/esp-idf-v5.1.2/components/freertos/FreeRTOS-Kernel/portable/xtensa/xtensa_vectors.S:564

I (299) cpu_start: Unicore app
I (300) cpu_start: Pro cpu up.
I (300) cpu_start: Single core mode
I (308) cpu_start: Pro cpu start user code
I (308) cpu_start: cpu freq: 240000000 Hz
I (308) cpu_start: Application information:
I (308) cpu_start: Project name:     BlueRetro
I (309) cpu_start: App version:      1
I (309) cpu_start: Compile time:     Mar 22 2024 16:36:16
I (310) cpu_start: ELF file SHA256:  3f7e64a0dddc6f3f...
I (311) cpu_start: ESP-IDF:          v5.1.2-dirty
I (312) cpu_start: Min chip rev:     v1.0
I (312) cpu_start: Max chip rev:     v3.99
I (313) cpu_start: Chip rev:         v1.0
I (313) cpu_start: App cpu up.
I (314) heap_init: Initializing. RAM available for dynamic allocation:
I (315) heap_init: At 3FFAFF10 len 000000F0 (0 KiB): DRAM
I (315) heap_init: At 3FFB6388 len 00001C78 (7 KiB): DRAM
I (316) heap_init: At 3FFB9A20 len 00004108 (16 KiB): DRAM
I (317) heap_init: At 3FFCD448 len 00012BB8 (74 KiB): DRAM
I (318) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (319) heap_init: At 3FFE4350 len 00003A48 (14 KiB): D/IRAM
I (319) heap_init: At 3FFE7E28 len 000181D8 (96 KiB): D/IRAM
I (320) heap_init: At 40078000 len 00008000 (32 KiB): IRAM
I (321) heap_init: At 4009E554 len 00001AAC (6 KiB): IRAM
I (322) heap_init: At 3FF80068 len 00001F98 (7 KiB): RTCRAM
I (324) spi_flash: detected chip: generic
I (324) spi_flash: flash io: dio
I (324) esp_core_dump_uart: Init core dump to UART
I (324) esp_core_dump_uart: Init core dump to UART
I (325) app_start: Starting scheduler on CPU0
I (326) main_task: Started on CPU0
I (326) main_task: Calling app_main()
# App main at 40083084
# App init at 40083094
# APP_CPU RESET: 0
# APP_CPU CLKGATE: 0
# APP_CPU STALL: 0
# APP_CACHE_CTRL: 00000030
# APP CPU STACK PTR: 3FFBC59C
# Hardcoded system : 24: OGX360
Guru Meditation Error: Core  0 panic'ed (InstrFetchProhibited). Exception was unhandled.

Core  0 register dump:
PC      : 0x00060034  PS      : 0x00060230  A0      : 0x80114339  A1      : xfbd0
A0  :03fd0 A1    : x0dee A2  :03ffb6a2c  A13     : 0x00000080
A14     : 0x8008180f  A15     : 0x3����⌂����

**Comment by konwektor** [comment]
did you clone mine repo, or  mixed some old versionfiles with new?

from first log
**I (282) cpu_start: ESP-IDF: v5.1.2-dirty**
esp idf 5.1.2 is changed somehow or copy not git clone.
**I (309) cpu_start: App version: 1**
need to show actual version (mine 1.9) - maybe i got something wrong with master - try cloning :
**git clone -b v1.9 https://github.com/konwektor/blueretro**
then 
**git submodule update  --init --recursive**
 then copy file 
**ogx360**
 from configs/wh1 into main directory and rename to **sdkconfig** then **idf.py build** 

at the end of first log:
**config_load_from_file: Upgrading cfg v2 to v3**
maybe some config files from other version are still there - not sure if "cfg v3 " is already from new version of darthclouds Blueretro with extra config file- this should not be here.
after config line comes panic error:
Guru Meditation Error: Core 0 panic'ed (LoadStoreAlignment). Exception was unhandled.

in second log same idf and app verson.

taken from esp idf docus:
 **If [IDF Monitor](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/tools/idf-monitor.html) is used, Program Counter values will be converted to code locations (function name, file name, and line number), and the output will be annotated with additional lines:

Core 0 register dump:
PC      : 0x400e14ed  PS      : 0x00060030  A0      : 0x800d0805  A1      : 0x3ffb5030
0x400e14ed: app_main at /Users/user/esp/example/main/main.cpp:36

A2      : 0x00000000  A3      : 0x00000001  A4      : 0x00000001  A5      : 0x3ffb50dc
A6      : 0x00000000  A7      : 0x00000001  A8      : 0x00000000  A9      : 0x3ffb5000
A10     : 0x00000000  A11     : 0x3ffb2bac  A12     : 0x40082d1c  A13     : 0x06ff1ff8
0x40082d1c: _calloc_r at /Users/user/esp/esp-idf/components/newlib/syscalls.c:51

A14     : 0x3ffb7078  A15     : 0x00000000  SAR     : 0x00000014  EXCCAUSE: 0x0000001d
EXCVADDR: 0x00000000  LBEG    : 0x4000c46c  LEND    : 0x4000c477  LCOUNT  : 0xffffffff

Backtrace: 0x400e14ed:0x3ffb5030 0x400d0802:0x3ffb5050
0x400e14ed: app_main at /Users/user/esp/example/main/main.cpp:36

0x400d0802: main_task at /Users/user/esp/esp-idf/components/esp32/cpu_start.c:470

To find the location where a fatal error has happened, look at the lines which follow the "Backtrace" line. Fatal error location is the top line, and subsequent lines show the call stack.**

So look in log for line with **Backtrace**:
**Backtrace: 0x400e14ed:0x3ffb5030 0x400d0802:0x3ffb5050
0x400e14ed: app_main at /Users/user/esp/example/main/main.cpp:36**
next line after will show  what line of code is causing problems.

If You mixing on purpose versions - Then I can not help because it can be 1000 reasons why new code not working - takes a lot of time to analize changes, and debugging them.
possible that is something wrong with yours esp idf - you can try to clone this branch:
**git clone git clone -b v5.1.2 --recursive https://github.com/espressif/esp-idf.git esp-idf-v5.1.2** then inside **esp-idf-v5.1.2** directory run:
**install.bat
export.bat** (cmd.exe window with administrator privileges).

Or download already compiled files from mine fork, in main page look to the right for **Releases**
https://github.com/konwektor/BlueRetro/releases/download/v1.9/Blueretro+Ogx360.v1.9.zip
use information provided in https://github.com/konwektor/BlueRetro/tree/master/!compiled

**Comment by konwektor** [comment]
I have made compilation from scratch with links from my last post. On my side is everything ok.

`W (29) boot: Unicore bootloader
I (271) cpu_start: Unicore app
I (271) cpu_start: Pro cpu up.
I (271) cpu_start: Single core mode
I (279) cpu_start: Pro cpu start user code
I (279) cpu_start: cpu freq: 240000000 Hz
I (279) cpu_start: Application information:
I (279) cpu_start: Project name:     BlueRetro
I (280) cpu_start: App version:      v1.9
I (281) cpu_start: Compile time:     Mar 23 2024 00:52:31
I (281) cpu_start: ELF file SHA256:  484302c6a8427608...
I (282) cpu_start: ESP-IDF:          v5.1.2
I (283) cpu_start: Min chip rev:     v1.0
I (283) cpu_start: Max chip rev:     v3.99
I (284) cpu_start: Chip rev:         v3.1
I (285) cpu_start: App cpu up.
I (285) heap_init: Initializing. RAM available for dynamic allocation:
I (286) heap_init: At 3FFAFF10 len 000000F0 (0 KiB): DRAM
I (287) heap_init: At 3FFB6388 len 00001C78 (7 KiB): DRAM
I (287) heap_init: At 3FFB9A20 len 00004108 (16 KiB): DRAM
I (288) heap_init: At 3FFCD4E8 len 00012B18 (74 KiB): DRAM
I (289) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (290) heap_init: At 3FFE4350 len 00003A48 (14 KiB): D/IRAM
I (291) heap_init: At 3FFE7E28 len 000181D8 (96 KiB): D/IRAM
I (291) heap_init: At 40078000 len 00008000 (32 KiB): IRAM
I (292) heap_init: At 4009E870 len 00001790 (5 KiB): IRAM
I (293) heap_init: At 3FF80068 len 00001F98 (7 KiB): RTCRAM
I (295) spi_flash: detected chip: generic
I (295) spi_flash: flash io: dio
I (296) esp_core_dump_uart: Init core dump to UART
I (296) esp_core_dump_uart: Init core dump to UART
I (297) app_start: Starting scheduler on CPU0
I (297) main_task: Started on CPU0
I (297) main_task: Calling app_main()
# App main at 40083090
# App init at 400830A0
# APP_CPU RESET: 0
# APP_CPU CLKGATE: 0
# APP_CPU STALL: 0
# APP_CACHE_CTRL: 00000030
# APP CPU STACK PTR: 3FFBC59C
# Hardcoded system : 24: OGX360
# bt_host_load_bdaddr_from_file: No BDADDR on SD. Using ESP32's MAC
I (357) BTDM_INIT: BT controller compile version [ec4ac65]
I (357) BTDM_INIT: Bluetooth MAC: e0:5a:1b:e3:5a:aa
I (367) phy_init: phy_version 4780,16b31a7,Sep 22 2023,20:42:16
Coex register Wi-Fi channel change btdm cb faild
# gid_update_sys: OGX360
# bt_hci_cmd_reset
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_local_features
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_local_version_info
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_bd_addr
# BT_HCI_EVT_CMD_COMPLETE
# local_name: BlueRetro_OGX360_5AAA
# local_bdaddr: E0:5A:1B:E3:5A:AA
# bt_hci_cmd_read_buffer_size
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_class_of_device
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_local_name
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_voice_setting
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_num_supported_iac
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_current_iac_lap
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_set_event_filter
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_conn_accept_timeout
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_supported_commands
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_ssp_mode
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_inquiry_mode
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_inquiry_rsp_tx_pwr_lvl
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_local_ext_features
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_stored_link_key
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_page_scan_activity
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_page_scan_type
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_le_host_supp
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_le_read_wl_size
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_le_clear_wl
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_le_add_dev_to_wl
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_delete_stored_link_key
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_class_of_device
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_local_name
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_set_event_filter
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_set_event_filter
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_auth_enable
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_set_event_mask
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_page_scan_activity
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_inquiry_scan_activity
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_page_scan_type
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_page_scan_timeout
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_hold_mode_act
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_scan_enable
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_default_link_policy
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_le_read_buffer_size
# BT_HCI_EVT_CMD_COMPLETE
# bt_att_set_le_max_mtu 247
# bt_hci_cmd_le_set_adv_param
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_le_set_adv_data
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_le_set_scan_rsp_data
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_le_set_adv_enable
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_le_set_scan_enable
# bt_hci_cmd_le_set_scan_param_active
# bt_hci_cmd_le_set_scan_enable
# bt_hci_cmd_periodic_inquiry
# BT_HCI_EVT_CMD_COMPLETE
# opcode: 0x200C error: 0x0C retry: 0
# bt_hci_cmd_le_set_scan_enable
# bt_hci_cmd_le_set_scan_param_active
# bt_hci_cmd_le_set_scan_enable
# bt_hci_cmd_periodic_inquiry
# BT_HCI_EVT_CMD_COMPLETE
# BT_HCI_EVT_CMD_COMPLETE
# BT_HCI_EVT_CMD_COMPLETE
# BT_HCI_EVT_CMD_COMPLETE
# BT_HCI_EVT_CMD_COMPLETE
# BT_HCI_EVT_CMD_COMPLETE
# BT_HCI_EVT_CMD_COMPLETE
# opcode: 0x0403 error: 0x0C retry: 0
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# mc_restore: restore sucessful!
I (887) gpio: GPIO[0]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
# internal_flag_init: External adapter
I (887) gpio: GPIO[2]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0
I (887) gpio: GPIO[4]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0
I (887) gpio: GPIO[12]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0
I (887) gpio: GPIO[15]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0
I (887) main_task: Returned from app_main()
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_CONN_REQUEST
# bt_hci_cmd_accept_conn_req
# Page dev: 0 type: 0 bdaddr: 8C:CD:E8:BB:6C:23
# BT_HCI_EVT_CMD_STATUS
# BT_HCI_EVT_ROLE_CHANGE
# BT_HCI_EVT_CONN_COMPLETE
# dev: 0 acl_handle: 0x0080
# bt_hci_cmd_le_set_adv_disable
# bt_hci_cmd_remote_name_request
# BT_HCI_EVT_CMD_COMPLETE
# BT_HCI_EVT_CMD_STATUS
# BT_L2CAP_CONN_REQ
# BT_L2CAP_CONF_REQ
# BT_HCI_EVT_REMOTE_NAME_REQ_COMPLETE:
# bt_type_update: dev: 0 type: 2 subtype: 6
# bt_hci_cmd_exit_periodic_inquiry
# bt_hci_cmd_le_set_scan_enable
# bt_hci_cmd_le_set_scan_param_passive
# bt_hci_cmd_le_set_scan_enable
# bt_hci_cmd_switch_role
# dev: 0 type: 2:6 Nintendo RVL-CNT-01-UC
# BT_L2CAP_CONF_REQ
# BT_L2CAP_CONN_REQ
# BT_HCI_EVT_CMD_COMPLETE
# BT_HCI_EVT_CMD_COMPLETE
# BT_HCI_EVT_CMD_COMPLETE
# BT_HCI_EVT_CMD_COMPLETE
# BT_HCI_EVT_CMD_STATUS
# BT_HCI_EVT_ROLE_CHANGE
# BT_L2CAP_CONF_REQ
# BT_L2CAP_CONF_RSP
# bt_hci_cmd_write_link_supervision_to
# bt_hid_wii_init
# bt_hid_cmd_wii_set_rep_mode
# BT_HCI_EVT_CMD_COMPLETE
# opcode: 0x0C37 error: 0x0C retry: 0
# BT_HIDP_WII_STATUS
OGX360_I2C: Init I2C
OGX360_I2C: Looking for attached OGX360 modules
OGX360_I2C: Player 0 Found
OGX360_I2C: Player 1 Found
OGX360_I2C: Player 2 Not Found
OGX360_I2C: Player 3 Not Found
`

**Reply by EdwardDanielSalazar** [reply]
interesting, I followed your instructions to the letter, and it already shows the version, although it continues to appear dirty, it still does not work, I wanted to make some changes in the code with respect to the sticks since apparently they have a small error, but it seems it won't be possible, I can't find why, I also thank you for the help, and your constant responses

**Reply by konwektor** [reply]
if you changed something in source files - compiler will know it - and add "dirty" to version number. - its normal.
You need to have clean version of ESP IDF - using dirty with changes somewhere in source code without github commits can make impossible to trace some problems. 
can you explain this small error in sticks code? you can also upload your code to git as "experimental" branch - it will be easier to see changes between and what is wrong.
